plugins/jobs/web_client/views/JobListWidget.js

```diff
@@ -90,99 +157,246 @@ var JobListWidget = View.extend({
+            }
+
+            tr.addClass('g-highlight');
+
+            window.setTimeout(function () {
+                tr.removeClass('g-highlight');
+            }, 1000);
+        } else {
+            this.render();
+        }
     },
-    _updateFilter: function (filter, newValues) {
-        // We need to work out what keys have been removed or added
-        // so we can update the filter. We do this rather than created
-        // a new filter inorder to preserve the existing user selections.
-        var currentValues = _.keys(filter), added, removed;
-        added = _.difference(newValues, currentValues);
-        removed = _.difference(currentValues, newValues);
-
-        _.each(added, function (value) {
-            filter[value] = true;
-        });
-        _.each(removed, function (value) {
-            delete filter[value];
+
+    _prepareDataForChart(jobs) {
+        let allRecords = [];
+        jobs.forEach(job => {
+            let id = job.get('_id');
+            let title = job.get('title');
+            let currentStatus = JobStatus.text(job.get('status'));
+            let updated = moment(job.get('updated')).format('L LT');
+            let records = job.get('segments')
+                .map(segment => {
+                    let status = segment.status;
+                    let elapsed = '';
+                    switch (status) {
+                        case 'Inactive':
+                            elapsed = -segment.elapsed;
+                            break;
+                        case 'Queued':
+                            elapsed = -segment.elapsed;
+                            break;
+                        default:
+                            elapsed = segment.elapsed;
+                    }
+                    return {
+                        id: id,
+                        title: title,
+                        updated: updated,
+                        status: status,
+                        currentStatus: currentStatus,
+                        elapsed: elapsed
+                    };
+                })
+                .filter(record => this.phasesFilter[record.status]);
+            if (records.length) {
+                allRecords = allRecords.concat(records);
+            } else {
+                allRecords.push({
+                    id: id,
+                    title: title,
+                    updated: updated,
+                    currentStatus: currentStatus
+                });
+            }
         });
+        return allRecords;
+    },
+
+    _fetchWithFilter() {
+        var filter = {};
+        if (this.userId) {
+            filter.userId = this.userId;
+        }
+        if (this.typeFilter) {
+            filter.types = JSON.stringify(this.typeFilter);
+        }
+        if (this.statusFilter) {
+            filter.statuses = JSON.stringify(this.statusFilter);
+        }
+        this.collection.params = filter;
+        this.collection.fetch({}, true);
```

Comment:
When the page first starts, the filter doesn't exist. I found those specific logics make more sense on the view than the model. So, I kept it in the view.